src/components/Questionnaire/QuestionnaireEditor.tsx

```diff
@@ -1200,10 +1206,21 @@ function QuestionEditor({
     unit,
   } = question;
 
+  // Memoize answer options to ensure unique IDs to avoid unnecessary re-renders in value field of AnwserOption
+
+  const annotatedAnswerOptions = useMemo(() => {
+    return (
+      answer_option?.map((option: any) => ({
+        ...option,
+        _id: option._id || crypto.randomUUID(),
+      })) || []
+    );
+  }, [answer_option]);
+
   const [expandedSubQuestions, setExpandedSubQuestions] = useState<Set<string>>(
     new Set(),
   );
-
+  const [inputValue, setInputValue] = useState("");
```

Comment:
the state's name could be more specific. currently is very ambiguous given the context of this code.